fiksi/src/constraints/expressions.rs

```diff
@@ -31,6 +32,61 @@ pub(crate) enum Expression {
     LineCircleTangency(LineCircleTangency),
 }
 
+pub(crate) struct VariableVariableEquality {
+    pub(crate) variable1_idx: u32,
+    pub(crate) variable2_idx: u32,
+}
+
+impl From<VariableVariableEquality> for Expression {
+    fn from(expression: VariableVariableEquality) -> Self {
+        Self::VariableVariableEquality(expression)
+    }
+}
+
+impl VariableVariableEquality {
+    /// See the note about inlining on [`PointPointDistance::compute_residual_and_gradient_`].
+    #[inline(always)]
+    fn compute_residual_and_gradient_(variables: [f64; 2]) -> (f64, [f64; 2]) {
+        let residual = variables[1] - variables[0];
+
+        let gradient = [-1., 1.];
```

Comment:
This is the neat part about these kinds of constraints. Conceptually, this is the singular/logical version of `PointPointDistance` (with `distance = 0`), which has a more complicated gradient. In fact, that constraint is numerically ill-behaved around 0.